temporalio/worker.py

```diff
@@ -698,6 +695,10 @@ async def _run_activity(
                     f"Failed building exception result: {inner_err}"
                 )
 
+        # If there is a manager here, we make it flush all pending heartbeats in
+        # the queue
+        if manager:
+            manager.flush_pending_heartbeats(task_token)
```

Comment:
This is required for multiprocess activities because it is technically possible some heartbeats remain in queue unprocessed